Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -1065,4 +1066,32 @@ theorem sigma_eq_iff_eq_comp_cast {α : Type*} {a b : Σii, Fin ii → α} :
     sigma_eq_of_eq_comp_cast _ h'⟩
 #align fin.sigma_eq_iff_eq_comp_cast Fin.sigma_eq_iff_eq_comp_cast
 
+section ExtendFun
+
+/-- Extend a function `f : Fin n → α` to a function on all natural numbers, by
+    defining `f i = a` for all `i ≥ n` and some given constant `a` -/
+@[simp]
+abbrev extendFun {α : Type*} {n : ℕ} (f : Fin n → α) (a : α) : ℕ → α :=
+  Subtype.extendFun (f ∘ equivSubtype.symm) (fun _ _ => a)
+
+@[simp] lemma extendFun_val' {α a n} (m) {f : Fin (n+m) → α} {i : Fin n} :
+    extendFun f a i.val = f (i.castAdd _) := by
+  apply Subtype.extendFun_of_p
+  apply Nat.lt_add_right _ _ _ i.isLt
+
+@[simp] lemma extendFun_comp_val' {α a n} (m) {f : Fin (n+m) → α} :
+    (extendFun f a) ∘ val = f ∘ (castAdd m) := by
+  funext i; apply extendFun_val'
+
+@[simp] lemma extendFun_add {α a n} (m) {f : Fin (n+(m+1)) → α} :
+    (extendFun f a) n = f ⟨n, by simp⟩ := by
```

Comment:
The statement of this one looks rather strange to me; I think it should instead be:
```
@[simp] lemma extendFun_apply_of_le {n} (f : Fin m → α) (a : α) (n : ℕ) (hn : n < m) :
    extendFun f a n = f ⟨n, hn⟩ := by
 ```